api/src/main/java/io/smallrye/stork/ServiceInstancesCache.java

```diff
@@ -0,0 +1,33 @@
+package io.smallrye.stork;
+
+import java.time.Duration;
+import java.time.LocalDateTime;
+import java.util.List;
+
+public class ServiceInstancesCache {
+
+    public static final Duration DEFAULT_REFRESH_INTERVAL = Duration.ofMinutes(5);
+    List<ServiceInstance> serviceInstances;
```

Comment:
why package-level access?

Reply:
I was thinking we could either have an abstract superclass for service discovery with caching, or a caching wrapper that would delegate refresh to the actual implementation

Reply:
Yes, I aggree. I was thinking the same at the end but finally commit like currently is. I will give a try this way.
For the visibility of serviceInstances etc, it was just a mistake, I will fix it